test/integration/depth_camera.cc

```diff
@@ -458,6 +463,229 @@ void DepthCameraTest::DepthCameraBoxes(
+    ASSERT_NE(depthCamera, nullptr);
+
+    ignition::math::Pose3d testPose(ignition::math::Vector3d(0, 0, 0),
+        ignition::math::Quaterniond::Identity);
+    depthCamera->SetLocalPose(testPose);
+
+    // Configure depth camera
+    depthCamera->SetImageWidth(imgWidth_);
+    EXPECT_EQ(depthCamera->ImageWidth(),
+      static_cast<unsigned int>(imgWidth_));
+    depthCamera->SetImageHeight(imgHeight_);
+    EXPECT_EQ(depthCamera->ImageHeight(),
+      static_cast<unsigned int>(imgHeight_));
+    depthCamera->SetFarClipPlane(farDist);
+    EXPECT_NEAR(depthCamera->FarClipPlane(), farDist, DOUBLE_TOL);
+    depthCamera->SetNearClipPlane(nearDist);
+    EXPECT_NEAR(depthCamera->NearClipPlane(), nearDist, DOUBLE_TOL);
+    depthCamera->SetAspectRatio(aspectRatio_);
+    EXPECT_NEAR(depthCamera->AspectRatio(), aspectRatio_, DOUBLE_TOL);
+    depthCamera->SetHFOV(hfov_);
+    EXPECT_NEAR(depthCamera->HFOV().Radian(), hfov_, DOUBLE_TOL);
+
+    depthCamera->CreateDepthTexture();
+    scene->RootVisual()->AddChild(depthCamera);
+
+    // Set a callback on the camera sensor to get a depth camera frame
+    float *scan = new float[imgHeight_ * imgWidth_];
+    ignition::common::ConnectionPtr connection =
+      depthCamera->ConnectNewDepthFrame(
+          std::bind(&::OnNewDepthFrame, scan,
+            std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+            std::placeholders::_4, std::placeholders::_5));
+
+    // rgb point cloud data callback
+    unsigned int pointCloudChannelCount = 4u;
+    float *pointCloudData = new float[
+        imgHeight_ * imgWidth_ * pointCloudChannelCount];
+    ignition::common::ConnectionPtr connection2 =
+      depthCamera->ConnectNewRgbPointCloud(
+          std::bind(&::OnNewRgbPointCloud, pointCloudData,
+            std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+            std::placeholders::_4, std::placeholders::_5));
+
+    // update and verify we get new data
+    g_depthCounter = 0u;
+    g_pointCloudCounter = 0u;
+    depthCamera->Update();
+    EXPECT_EQ(1u, g_depthCounter);
+    EXPECT_EQ(1u, g_pointCloudCounter);
+
+    double expectedDepth = boxPosition.X() - boxSize.X() * 0.5;
+
+    double pointAvg = 0.0;
+    double depthAvg = 0.0;
+    // Verify depth and point cloud data before particle effects
+    for (unsigned int i = 0; i < depthCamera->ImageHeight(); ++i)
+    {
+      unsigned int step = i*depthCamera->ImageWidth()*pointCloudChannelCount;
+      for (unsigned int j = 0; j < depthCamera->ImageWidth(); ++j)
+      {
+        float x = pointCloudData[step + j*pointCloudChannelCount];
+        float y = pointCloudData[step + j*pointCloudChannelCount + 1];
+        float z = pointCloudData[step + j*pointCloudChannelCount + 2];
+        EXPECT_NEAR(expectedDepth, x, DEPTH_TOL);
+        float d = scan[i * depthCamera->ImageWidth() + j];
+        EXPECT_NEAR(expectedDepth, d, DEPTH_TOL);
```

Comment:
Can we use `EXPECT_DOUBLE_EQ` instead of `EXPECT_NEAR`?

Reply:
the depth values are generated by reading the depth buffer and this has some precision issue so it's using `DEPTH_TOL` which has a higher tolerance.